src/main.rs

```diff
@@ -538,10 +558,13 @@ fn run_cli(cli: Cli) -> Result<()> {
                 container.drain(),
                 &mut record_db,
                 &client,
+                format,
                 ignore_null,
             );
 
-            output_entries(out.as_ref(), valid_entries)?;
+            if format != OutputFormat::Null {
+                output_entries(out.as_ref(), valid_entries)?;
+            }
```

Comment:
I would probably prefer if the `write_entries` / `output_entries` signature were modified to include the `OutputFormat` as an argument, to make it easier to implement new formats in the future.